qiskit/aqua/operators/primitive_ops/matrix_op.py

```diff
@@ -108,21 +110,62 @@ def equals(self, other: OperatorBase) -> bool:
             return self.coeff == other.coeff and self.primitive == other.primitive
         return self.coeff * self.primitive == other.coeff * other.primitive  # type: ignore
 
+    def expand_with_identity(self, num_qubits: int) -> 'MatrixOp':
+        identity = np.identity(2**num_qubits, dtype=complex)
+        return MatrixOp(self.primitive.tensor(Operator(identity)), coeff=self.coeff)  # type: ignore
+
     def tensor(self, other: OperatorBase) -> OperatorBase:
         if isinstance(other.primitive, Operator):  # type: ignore
             return MatrixOp(self.primitive.tensor(other.primitive),  # type: ignore
                             coeff=self.coeff * other.coeff)  # type: ignore
 
         return TensoredOp([self, other])
 
-    def compose(self, other: OperatorBase) -> OperatorBase:
-        other = self._check_zero_for_composition_and_expand(other)
+    def compose(self, other: OperatorBase,
+                permute_self: List[int] = None,
+                permute_other: List[int] = None) -> OperatorBase:
 
+        self, other = self._check_zero_for_composition_and_expand(other,  # type: ignore
```

Comment:
Could you add a comment that this does the proper resorting? It's not very obvious from the method name

Reply:
maybe we can rename the method to ```_expand_the_shorter_and_permute```?
check_zero functionality only expands the shorter operator if it's Zero